app/validators/questionnaire_validator.py

```diff
@@ -94,3 +99,19 @@ def validate_smart_quotes(self):
                         error_messages.DUMB_QUOTES_FOUND,
                         pointer=translatable_item.pointer,
                     )
+
+    def validate_introduction_block(self):
+        blocks = self.questionnaire_schema.get_blocks()
+        has_introduction_blocks = any(
+            block["type"] == "Introduction" for block in blocks
+        )
+        if not has_introduction_blocks:
+            self.add_error(error_messages.NO_INTRODUCTION_BLOCK)
+
+    def validate_section_title(self):
```

Comment:
This should be done via JSON schema validation.

Reply:
I tried but since we need `questionnaire_flow` field and section's nested attributes is it possible in JSON schema validation? I can't find similar example.